src/Importer.lua

```diff
@@ -85,13 +135,24 @@ function Importer:getNextInstance(current, pathPart, hasAscendedParents)
 			return current.Parent.Parent
 		end
 	else
-		local alias = self._config.aliases[pathPart]
-
-		if alias then
-			return alias
-		else
-			return current:FindFirstChild(pathPart)
+		if isFirstPart then
+			local alias = self._config.aliases[pathPart]
+			local isService = self.dataModel == game and checkIfIsService(pathPart)
+
+			if alias then
+				if isService then
+					warn("Import: Alias '%s' is also defined as a Roblox service. Using alias instead.")
+				end
+
+				return alias
+			elseif isService then
+				return game:GetService(pathPart)
+			else
+				return self:getChild(self.dataModel, pathPart)
+			end
 		end
+
+		return self:getChild(current, pathPart)
```

Comment:
This tree is getting a little hairy. I don't think we need to do anything about it right now, but as the project goes we should be careful about how much more complexity we introduce here, as it will get progressively harder to reason about.